cmake/modules/IsStaticLibrary.cmake

```diff
@@ -0,0 +1,41 @@
+#[=======================================================================[.rst:
+IsStaticLibrary
```

Comment:
This module is a good idea. I am wondering if maybe we should invert it to IsSharedLibrary to be consistent with the standard CMake variable BUILD_SHARED_LIBS.

Reply:
This checks explicit for static, this is what we want to know.

Reply:
okay